backend/data/processors/conditions_preprocessing.py

```diff
@@ -518,5 +524,118 @@ def unsw_global_degrees(processed):
 
 
 
+def remove_extraneous_comm_data(processed):
+    """Adds handbook notes and removes phrasing
+    example: """
+    note = ""
+    # Academic Standing note
+    processed, number_of_subs = re.subn(
+        r"Students must be in Good Academic Standing", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        note += "Students must be in Good Academic Standing."
+
+    processed, number_of_subs = re.subn(
+        r"on Good Standing and", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        note += "Students must be in Good Academic Standing."
+
+    processed, number_of_subs = re.subn(
+        r"Good Standing", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        note += "Students must be in Good Academic Standing."
+
+
+    # Final 2 terms note
+    processed, number_of_subs = re.subn(
+        r"It is strongly recommended that students only complete this course within their final 2 terms of study", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        note += "Recommended to take this course within their final 2 terms of study."
+
+    # Business school students terms notes
+    terms_string = re.findall("Only available to single and double degree Business School students in Term \d. It will be offered to non-Business School students in Terms \d and \d.", processed)
+
+    processed, number_of_subs = re.subn(
+        r"Only available to single and double degree Business School students in Term \d. It will be offered to non-Business School students in Terms \d and \d.", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        terms_numbers = re.findall(r'[0-9]+', terms_string[0])
+        note += r"Only available to single and double degree Business School students in Term " + terms_numbers[0] + ". It will be offered to non-Business School students in Terms " + terms_numbers[1] +" and " + terms_numbers[2] +"."
+
+    processed, number_of_subs = re.subn(
+    r"Only available to single and double degree Business School students in Term 2. It will be offered to non-Business School students in Term 3.", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
+        note += "Only available to single and double degree Business School students in Term 2. It will be offered to non-Business School students in Term 3."
+
+    # Application and progression check notes
+    processed, number_of_subs = re.subn(
+    r"This course is by application only. Please contact the Co-op office for more information", r"", processed, flags=re.IGNORECASE
+    )
+    if number_of_subs > 0:
```

Comment:
this pattern of `if number_of_subs` is a little repetitive, you should make a general function to handle that logic.

```
def add_note_if_found(processed, find, note, replace_with):
    processed, number_of_subs = re.subn(find, r"", processed, flags=re.IGNORECASE)
    return processed, note + replace_with if number_of_subs > 0 else processed, note
```

you can call this function like:
```
processed, note = add_note_if_found(processed, Only available to single and double degree Business School students in Term " + terms_numbers[0] + ". It will be offered to non-Business School students in Terms " + terms_numbers[1] +" and " + terms_numbers[2] +".", note, Only available to single and double degree Business School students in Term 2. It will be offered to non-Business School students in Term 3.")
```